Fix: Refresh sgrid on figure resize issue #15

nilesh646 opened the pull request:
This pull request fixes issue #28 by ensuring that the sgrid updates when the figure is resized.

Changes:
- Added a resize callback to update the sgrid dynamically
- Tested with:
  %! hfig = get(0, "currentfigure");
  %! hax = get(hfig, "currentaxes");
  %! sgrid(hax, "on");

Let me know if any additional tests or refinements are required.

ttl-octave reviewed (changes requested):
Thank you very much for your contribution, very much appreciated. I have added some comments which should be addressed before merging.
And just to be sure (as there is no Octave-specific syntax in the changes like line comments or end statements): Have you written the code or is it copied from somewhere?

nilesh646 commented:
Thank you for the detailed review!
I’ve addressed all comments:
-  Confirm that the code is my original work, although I referred to existing implementations for guidance
- Matched indentation style.
- Replaced `guidata` with `userdata`.
- Used Octave-specific `endif`, `endfunction`.
- Updated logic to preserve custom zeta/omega values.

Let me know if anything else is needed.

nilesh646 commented:
Thank you for pointing that out! I've corrected the field names and updated the logic to call __sgrid_create__ with the appropriate arguments. I've also added the missing endif to close the conditional block properly.

Please let me know if there’s anything else you'd like me to adjust.

ttl-octave commented:
Thanks for solving issue #15!
